src/cfront.c

```diff
@@ -1269,6 +1280,35 @@ void read_lvalue(lvalue_t *lvalue,
     }
 }
 
+void read_ternary_operation(int dest, block_t *parent)
+{
+    if (lex_accept(T_question)) {
```

Comment:
You can rewrite with false-condition-first manner. That is,
```c
if (!lex_accept(T_question)) return;

/* the reset code */
```

Then, the program can be shortened.